mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,394 @@
+- The  overridden ``Process`` method controls what the tag helper does when executed. The ``TagHelper`` class also provides an asynchronous version (``ProcessAsync``) with the same parameters.
+- The context parameter to ``Process`` (and ``ProcessAsync``) contains information associated with the execution of the current HTML tag.
+- The output parameter to ``Process`` (and ``ProcessAsync``) contains a stateful HTML element representative of the original source used to generate an HTML tag and content.
+- Our class name has a suffix of **TagHelper**, which is *not* required, but it's considered a best practice convention. You could declare the class as:
+
+.. code-block:: c#
+
+	public class Email : TagHelper
+
+And even without the ``[TargetElement]`` attribute, ``<email>`` tags will be targeted.
+
+2. To make the ``EmailTagHelper`` class available to all our Razor views, we will add the ``addTagHelper`` command to the *Views/_ViewImports.cshtml* file:
+
+.. wildcard syntax
+
+.. literalinclude:: authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/Views/_ViewImportsCopy.cshtml
+   :language: aspx-cs
+   :emphasize-lines: 2,3
+
+The code above uses the wildcard syntax to specify all the tag helpers in our DLL will be available. The first string after ``@addTagHelper`` specifies the tag helper to load (we are using "\*" for all tag helpers), and the second string "AuthoringTagHelpers" specifies the DLL the tag helper is in. Also, note that the second line brings in the ASP.NET 5 MVC 6 tag helpers using the wildcard syntax (those helpers are discussed in :doc:`intro`.) It's the ``@addTagHelper`` command that makes the tag helper available to the Razor view. Alternatively, you can provide the fully qualified name (FQN) of a tag helper as shown below:
+
+.. FQN syntax
+
+.. literalinclude:: authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/Views/_ViewImports.cshtml
+   :language: aspx-cs
+   :lines: 1-3
+   :emphasize-lines: 3
+
+To add a tag helper to a view using a FQN, you first add the FQN (``AuthoringTagHelpers.TagHelpers.EmailTagHelper``), and then the DLL name (*AuthoringTagHelpers*). Most developers will prefer to use the wildcard syntax. :doc:`using` goes into detail on tag helper adding, removing, hierarchy, and wildcard syntax. 
+
+3. Update the markup in the *Views/Home/Contact.cshtml* file with these changes:
+
+.. literalinclude::  authoring/sample/AuthoringTagHelpers/src/AuthoringTagHelpers/Views/Home/Contact.cshtml
+   :language: aspx-cs
+   :emphasize-lines: 15-16
+   :lines: 1-17
+   
+4. Run the app and use your favorite browser to view the HTML source so you can verify that the email tag is replaced with anchor markup (``<a>Support</a>``). Support is rendered as a link, but it doesn't have an ``href`` attribute to make it functional. We'll fix that in the next section.
```

Comment:
You also had the user add a `<email>Marketing</email>`. Therefore, we should ask for the user to verify both the support and marketing bits.